pennylane/utils.py

```diff
@@ -105,6 +106,56 @@ def decompose_hamiltonian(H, hide_identity=False):
     return coeffs, obs
 
 
+def sparse_hamiltonian(H):
+    r"""Computes the sparse matrix representation a Hamiltonian in the computational basis.
+
+    Args:
+        H (~.Hamiltonian): Hamiltonian operator for which the matrix representation should be
+         measured
+
+    Returns:
+        coo_matrix: a sparse matrix in scipy COOrdinate format with the dimension of
+        :math:`(2^n, 2^n)` where :math:`n` is the number of wires
+
+    **Example:**
+
+    This function can be used by passing a `qml.Hamiltonian` object as:
+
+    >>> coeffs = [0.5, 0.5]
+    >>> obs = [qml.PauliZ(wires=[0]) @ qml.PauliZ(wires=[1]),
+    ...        qml.Identity(wires=[0]) @ qml.PauliZ(wires=[1])]
+    >>> H = qml.Hamiltonian(coeffs, obs)
+    >>> H_sparse = sparse_hamiltonian(H)
+
+    The resulting sparse matrix can be either used directly or transformed into a numpy array:
```

Comment:
It's good to print the sparse version too, since maybe not all users are familiar with the COO format